### PR TITLE
Feature/mcu_module request ecu status

### PR DESCRIPTION
**Pull request by alexandrudoncea**
- further sync with ECU GenerateFrames
- method to send requests to ECUs to check if it is UP -> not implemented yet

### CODE REVIEW

**Review by GeorgeCimpoeru — APPROVED** [review]

File: src/mcu/Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+FINAL = main
+FINAL_TEST = test_binary
+.PHONY: all clean
+
+all: $(OBJ_DIR) $(FINAL) 
+$(FINAL): $(OBJS)
+	$(CXX) $(CFLAGS) -o $(FINAL) $(OBJS) ${LDFLAGS}
+$(OBJ_DIR):
+	mkdir -p $(OBJ_DIR)
+$(OBJ_DIR)/main.o: src/main.cpp include/MCUModule.h include/ReceiveFrames.h include/HandleFrames.h include/GenerateFrame.h include/InterfaceModule.h
```
Comment (Miruna-Maria) [inline src/mcu/Makefile:21]: same here

File: src/mcu/Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	$(CXX) $(CFLAGS) -c src/main.cpp -o $(OBJ_DIR)/main.o
+$(OBJ_DIR)/MCUModule.o: src/MCUModule.cpp
+	$(CXX) $(CFLAGS) -c src/MCUModule.cpp -o $(OBJ_DIR)/MCUModule.o
+$(OBJ_DIR)/ReceiveFrames.o: src/ReceiveFrames.cpp
+	$(CXX) $(CFLAGS) -c src/ReceiveFrames.cpp -o $(OBJ_DIR)/ReceiveFrames.o
+$(OBJ_DIR)/HandleFrames.o: src/HandleFrames.cpp
+	$(CXX) $(CFLAGS) -c src/HandleFrames.cpp -o $(OBJ_DIR)/HandleFrames.o
+$(OBJ_DIR)/GenerateFrame.o: src/GenerateFrame.cpp
```
Comment (Miruna-Maria) [inline src/mcu/Makefile:29]: here also

File: src/mcu/Makefile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	$(CXX) $(CFLAGS) -c src/GenerateFrame.cpp -o $(OBJ_DIR)/GenerateFrame.o
```
Comment (Miruna-Maria) [inline src/mcu/Makefile:30]: here also

File: src/mcu/include/GenerateFrames.h [hunk 4]
```diff
@@ -32,7 +32,7 @@ enum FrameType
     OVERLOAD_FRAME
```
Comment (Miruna-Maria) [inline src/mcu/include/GenerateFrames.h:32]: i cannot comment above, please modify in comment GenerateFrames.h on line 2

File: src/mcu/src/GenerateFrames.cpp [hunk 5]
```diff
@@ -1,13 +1,13 @@
 /**
- * @file GenerateFrame.cpp
```
Comment (Miruna-Maria) [inline src/mcu/src/GenerateFrames.cpp:2]: please modify with GenerateFrames.cpp

File: src/mcu/src/GenerateFrames.cpp [hunk 6]
```diff
@@ -423,45 +424,49 @@ void GenerateFrame::TransferDataLong(uint32_t can_id, uint8_t block_sequence_cou
...
 }
 
-void GenerateFrame::InsertBytes(std::vector<uint8_t>& data, uint8_t index, uint8_t num_bytes) 
+void GenerateFrames::insertBytes(std::vector<uint8_t>& data, unsigned int index, int num_bytes) 
 {
-    for (uint8_t bit = 0; bit < num_bytes; bit++) 
+    // for (uint8_t bit = 0; bit < num_bytes; bit++) 
+    // {
+    //     data.insert(data.begin() + index, 0x00);
+    // }
```
Comment (Miruna-Maria) [inline src/mcu/src/GenerateFrames.cpp:462]: these can be deleted

**Review by RubenNatanael — APPROVED** [review]

**Pull request merged**